lib/Doctrine/Common/Collections/Collection.php

```diff
@@ -48,24 +45,6 @@ public function add($element);
      */
     public function clear();
```

Comment:
Should this one be moved up as well? Calling it still preserves the guarantee that the type of the items stays the same.

Reply:
this one would not really fit the definition of `Readable` though. So I would keep it here even though it is not a blocker for the covariance of the template type.